### PR TITLE
Added Zookeeper charm.

### PR DESCRIPTION
**Pull request by pengale**
Based on the vanilla apache-zookeeper charm, but utlizes Bigtop to
reduce complexity in the charm itself.

Note that this depends on https://github.com/juju-solutions/layer-apache-bigtop-base/pull/18

The charm won't work until that is part of our base layer.

@juju-solutions/bigdata

### CODE REVIEW

File: bigtop-deploy/juju/hadoop-zookeeper/reactive/zookeeper.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    nodes = zkpeer.get_nodes()  # single node since we dismiss .departed below
+    zookeeper = Zookeeper()
+    zookeeper.decrease_quorum(nodes)
+    zkpeer.dismiss_departed()
+    hookenv.status_set('active', 'Ready')
+
+
+@when('zookeeper.started', 'zkclient.joined')
+def serve_client(client):
+    config = DistConfig()
```
Comment (pengale) [inline bigtop-deploy/juju/hadoop-zookeeper/reactive/zookeeper.py]: Whoops. This line is wrong. I need to either point DistConfig at the base layer's config, as in lib/.../zookeeper.py, or get the ports out of a different config file. (dist.yaml doesn't exist in this charm.)

**Comment by pengale** [comment]
Moved this to the right place (I hope this is the right place ...)

File: bigtop-packages/src/charm/hadoop/layer-hadoop-zookeeper/reactive/zookeeper.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+@when_not('zookeeper.installed')
+def install_zookeeper():
+    '''
+    After Bigtop has done the initial setup, trigger a puppet install,
+    via our Zooekeeper library.
+
+    puppet will start the service, as a side effect.
+
+    '''
+    hookenv.status_set('waiting', 'Installing Zookeeper')
```
Comment (ktsakalozos) [inline bigtop-packages/src/charm/hadoop/layer-hadoop-zookeeper/reactive/zookeeper.py]: I am sorry about that I misled you. When we are doing things within the charm we use 'maintenance' and not 'waiting' eg hookenv.status_set('maintenance', 'installing namenode')

Sorry again...

**Comment by pengale** [comment]
Closing this PR for now. Will recreate, with properly formatted title, and a branch name matching our convention, after I've added smoke tests.

**Pull request closed without merging**